cache/models/subject.js

```diff
@@ -186,4 +201,114 @@ function completeSubjectHierarchy(res, params) {
 
 module.exports = {
   completeSubjectHierarchy,
+
+  /**
+   * Returns subject with filter options if provided.
+   * @param  {Object} req - Request object
+   * @param  {Object} res - Result object
+   * @param  {Object} logObject - Log object
+   * @returns {Promise} - Resolves to a subject objects
+   */
+  getSubject(req, res, logObject) {
+    const opts = modelUtils.getOptionsFromReq(req.swagger.params, helper);
+    const key = sampleStore.toKey(constants.objectType.subject, opts.filter.key);
+    return redisClient.hgetallAsync(key)
+    .then((subject) => {
+      if (!subject) {
+        throw new redisErrors.ResourceNotFoundError({
+          explanation: 'Sample not found.',
+        });
+      }
+
+      // add parentAbsolutePath
+      if (subject.parentAbsolutePath === undefined) {
+        subject.parentAbsolutePath = '';
+      }
+
+      // convert the strings into numbers
+      subject.childCount = parseInt(subject.childCount, 10) || 0;
```

Comment:
On master :
GET ALL subjects : the numeric fields: childCount, hierarchyLevel, are stringified and the parentAbsolutePath of root subject is undefined. https://refocus-devbox.herokuapp.com/v1/subjects?name=Salesforce

For GET subject: the same numeric fields are numbers, and for root subjects parentAbsolutePath=''. https://refocus-devbox.herokuapp.com/v1/subjects/Salesforce

This PR reproduces these inconsistencies. We can discuss what is the best course of action after I get back.